proxy-server/proxy-server.go

```diff
@@ -218,7 +218,7 @@ func handleRequestAndRedirect(res http.ResponseWriter, req *http.Request) {
 		if err != nil || !tkn.Valid {
 			log.Printf("Authorization Error: invalid token from saturn")
 			// Breaking the loop. No more redirects. All is bad for this request.
-			res.WriteHeader(http.StatusUnauthorized)
+			respondWithError(res, http.StatusUnauthorized, "Invalid token.")
```

Comment:
Aren't we being too nice to give away cause of the authorization failure? :)

Reply:
It's the same message for all cases - not telling them what specifically is wrong with the token :). I figured this was cryptic enough for a customer to go "this is wrong" and either try again or contact support.